packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts

```diff
@@ -0,0 +1,339 @@
+import { IPanelStyleProps, IPanelStyles, PanelType } from './Panel.types';
+import {
+  AnimationClassNames,
+  AnimationVariables,
+  FontSizes,
+  IconFontSizes,
+  getGlobalClassNames,
+  ScreenWidthMinMedium,
+  ScreenWidthMinXLarge,
+  ScreenWidthMinXXLarge,
+  ScreenWidthMinUhfMobile
+} from '../../Styling';
+import { IStyleFunctionOrObject } from '../../Utilities';
+import { IIconStyles, IIconStyleProps } from '../../Icon';
+
+const GlobalClassNames = {
+  root: 'ms-Panel',
+  main: 'ms-Panel-main',
+  commands: 'ms-Panel-commands',
+  contentInner: 'ms-Panel-contentInner',
+  navigation: 'ms-Panel-navigation',
+  closeButton: 'ms-Panel-closeButton ms-PanelAction-close',
+  header: 'ms-Panel-header',
+  headerText: 'ms-Panel-headerText',
+  content: 'ms-Panel-content',
+  footer: 'ms-Panel-footer',
+  footerInner: 'ms-Panel-footerInner',
+  isOpen: 'is-open',
+  hasCloseButton: 'ms-Panel--hasCloseButton',
+  smallFluid: 'ms-Panel--smFluid',
+  smallFixedNear: 'ms-Panel--smLeft',
+  smallFixedFar: 'ms-Panel--sm',
+  medium: 'ms-Panel--md',
+  large: 'ms-Panel--lg',
+  largeFixed: 'ms-Panel--fixed',
+  extraLarge: 'ms-Panel--xl',
+  custom: 'ms-Panel--custom'
+};
+
+function getIconStyles(props: IPanelStyleProps): IStyleFunctionOrObject<IIconStyleProps, IIconStyles> {
+  const { theme } = props;
+  return () => ({
+    root: {
+      height: 'auto',
+      width: '44px',
+      color: theme.palette.neutralSecondary,
+      fontSize: IconFontSizes.large
+    },
+    rootHovered: {
+      color: theme.palette.neutralPrimary
+    }
+  });
```

Comment:
Actually I'm a bit confused now. I don't see `rootHovered` in `IIconStyles`, so I'm not sure how this would work. It is in `IButtonStyles`. Is `icon` styles an `Icon` or an `IconButton`? Should the return type be `IButtonStyles` instead?

Also, I don't actually see `styles.icon` being passed to the `IconButton` in `Panel.base` (or anywhere else.) Is it being applied anywhere?

Reply:
+1 to Jason's comments about simply using an object.